packages/amplify-graphql-default-value-transformer/src/graphql-default-value-transformer.ts

```diff
@@ -31,16 +31,24 @@ import { TypeValidators } from './validators';
 
 const nonStringTypes = ['Int', 'Float', 'Boolean', 'AWSTimestamp', 'AWSJSON'];
 
-const validateFieldType = (ctx: TransformerSchemaVisitStepContextProvider, type: TypeNode): void => {
+const validateFieldType = (ctx: TransformerSchemaVisitStepContextProvider, config: DefaultValueDirectiveConfiguration): void => {
+  const type = config.field.type;
+  const argc = config.directive.arguments!.length;
   const enums = ctx.output.getTypeDefinitionsOfKind(Kind.ENUM_TYPE_DEFINITION) as EnumTypeDefinitionNode[];
   if (isListType(type) || !isScalarOrEnum(type, enums)) {
     throw new InvalidDirectiveError('The @default directive may only be added to scalar or enum field types.');
   }
+  if (isPostgresModel(ctx, config.object.name.value) && argc === 0 && getBaseType(type) !== 'Int') {
+    throw new InvalidDirectiveError('The empty @default (auto-increment) may only be applied to integer fields.');
```

Comment:
see note above re error message. Also, maybe say "fields with Int type" or similar to make it clear that you're talking about the GraphQL type `Int`.